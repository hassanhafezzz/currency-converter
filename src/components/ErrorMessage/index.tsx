import React, { FC } from 'react';
import classNames from 'classnames/bind';

import styles from './styles.module.css';

const cx = classNames.bind(styles);

const ErrorMessage: FC<{ dismissMessage: () => void }> = ({
  dismissMessage,
  children,
}) => {
  return (
    <div role="alert" className={cx('error')}>
      <button type="button" className={cx('close')} onClick={dismissMessage}>
        close
      </button>
      <p>{children}</p>
    </div>
  );
};

export default ErrorMessage;
