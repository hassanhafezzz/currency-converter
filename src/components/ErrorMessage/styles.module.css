.error {
  display: flex;
  flex-direction: column;
  position: fixed;
  bottom: 2rem;
  left: 50%;
  transform: translateX(-50%);
  color: var(--color-red-500);
  background-color: var(--color-red-100);
  padding: 1rem 0.2rem;
  font-size: var(--small-paragraph);
  border-radius: 8px;
  text-align: center;
  filter: drop-shadow(0 20px 20px var(--color-primary-900));
}

.error p {
  margin: 0.7rem 0 1rem 0;
  padding: 0 2rem;
}

.close {
  padding: 0.2rem 0.5rem;
  margin-right: 1rem;
  outline: none;
  border: none;
  background-color: var(--color-red-500);
  border-radius: 2rem;
  color: var(--color-red-100);
  font-size: var(--small-paragraph);
  text-transform: capitalize;
  cursor: pointer;
  align-self: flex-end;
}
