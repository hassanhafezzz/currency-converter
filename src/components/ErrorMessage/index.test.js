import React from 'react';
import { render, screen } from '@testing-library/react';
import ErrorMessage from './index';
import user from '@testing-library/user-event';

test('Error displayed correctly and close button is clickable', async () => {
  const message = 'error message';
  const mockClose = jest.fn();
  render(<ErrorMessage dismissMessage={mockClose}>{message}</ErrorMessage>);
  expect(screen.getByText(message)).toBeInTheDocument();

  user.click(await screen.findByText(/close/i));

  expect(mockClose).toHaveBeenCalledTimes(1);
});

test('Add snapshot for error messages ', () => {
  const { container, asFragment } = render(<ErrorMessage />);
  expect(asFragment(container)).toMatchSnapshot();
});
