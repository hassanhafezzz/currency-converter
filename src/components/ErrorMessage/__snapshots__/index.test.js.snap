// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add snapshot for error messages  1`] = `
<DocumentFragment>
  <div
    class="error"
    role="alert"
  >
    <button
      class="close"
      type="button"
    >
      close
    </button>
    <p />
  </div>
</DocumentFragment>
`;
